psi4/src/psi4/cc/ccresponse/scatter.cc

```diff
@@ -408,7 +408,7 @@ void scatter(std::shared_ptr<Molecule> molecule, Options &options, double step,
     for (i = 0; i < natom; i++) {
         for (j = 0; j < 3; j++) {
             // M->set((i*3+j),(i*3+j),1/sqrt(an2mass[(int)zvals[i]]/_au2amu));
-            M->set((i * 3 + j), (i * 3 + j), 1 / sqrt((molecule->mass(i)) / pc_au2amu));
+            M->set((i * 3 + j), (i * 3 + j), 1 / sqrt((molecule->mass(i, false)) / pc_au2amu));
```

Comment:
We will have to be careful, if atom `i` here is a ghost and the mass is zero this would cause a divide by zero error. I don't think that coordinates for ghost atoms should be included in the cartesian hessian? We might need @lothian to chime in here.

Reply:
I believe we're in agreement with @lothian that the cartesian hessian should not include ghost atoms. However, a quick test of `psi4.hessian` on Helium with and without adding a Gh(He) reveals that it does indeed include the ghost atom (goes from size 3x3 to 6x6); so, skipping ghosts here would create a mass-weighting matrix of the "wrong" size (really a hessian with the wrong size) which should throw an error below at the `gemm` when  mass-weighting.

The question is- should we fix the hessian and `scatter.cc` here and now, should we only fix `scatter.cc` and leave the error at the `gemm` and fix the hessian in a separate PR, or let sleeping ghosts lie? I'm not particularly fond of the latter.

Reply:
The Hessian should not include ghost atoms at all, in my opinion.